docs/resource-providers/index.md

```diff
@@ -40,7 +40,22 @@ Example implementations:
 class VpcResourceProvider implements ResourceProvider<IVpc> {
     provide(context: ResourceContext): IVpc {
         const scope = context.scope; // stack
-        ...
```

Comment:
I would rather add this is as a separate example that over writing on existing one.